app/cafe.py

```diff
@@ -0,0 +1,33 @@
+import datetime
+from app.errors import (
+    NotVaccinatedError,
+    OutdatedVaccineError,
+    NotWearingMaskError
+)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if not isinstance(visitor, dict):
+            raise NotVaccinatedError("Visitor is incorrect and not vaccinated")
+        visitor_name = visitor.get("name")
+        visitor_vaccine_dict = visitor.get("vaccine")
+
+        if not visitor_vaccine_dict:
+            raise NotVaccinatedError(f"Visitor {visitor_name} "
+                                     f"is not vaccinated")
+
+        vaccine_date = visitor_vaccine_dict.get("expiration_date")
+        if (not isinstance(vaccine_date, datetime.date)
+                or vaccine_date < datetime.date.today()):
+            raise OutdatedVaccineError(f"The vaccine of {visitor_name} "
+                                       f"is expired")
```

Comment:
The error message could be more descriptive. Instead of `The vaccine of {visitor_name} is expired`, it would be better to say `The vaccination of {visitor_name} is expired`, because the current message implies that the vaccine itself is expired, while it's actually about the expiration of the vaccination effect.